pysher/pusher.py

```diff
@@ -166,6 +169,9 @@ def _generate_presence_token(self, channel_name):
         :rtype: str
         """
         subject = "{}:{}:{}".format(self.connection.socket_id, channel_name, json.dumps(self.user_data))
+        if not isinstance(subject, bytes):
```

Comment:
This check isn't needed. It is *always* a string, since we're constructing it just in the line above.
replace this with the following here and above
```
subject = subject.encode('utf-8')
```

and this is good to be merged.

Reply:
I'm not checking if it's `str`, I'm checking if it's `bytes`. On Python 2, `subject` is of type `bytes`. If you do `subject.encode('utf-8')` there, it will first try to decode the string to unicode using `ascii` encoding and then re-encode it to `utf-8`. This process will fail with `UnicodeDecodeError` if there are any non-ascii characters in the string.

Reply:
This library doesn't support python <3.5. Hence, this check isn't needed as it's always a string.

Reply:
Ah, my mistake.